### DISCUSSION TITLE
Why isnt this simple component working?

### DISCUSSION THREAD

**Discussion by viezel** [alpinejs/alpine#615] [4. General]
Im doing a super basic component to learn Alpine. Idea is to fetch remote data and show it once completed. I added a preloader and an error div to complete it a bit more.

but the loading div is never removed - why is that?

```javascript
<div x-data="{
    rows: [],
    hasError: false,
    isLoading: false,
    fetchData() {
        this.isLoading = true;
        fetch('https://jsonplaceholder.typicode.com/users')
        .then(response => response.json())
        .then(function (data) {
            this.rows = data;
        })
        .catch(function (error) {
            this.hasError = true;
        })
        .then(function () {
            this.isLoading = false;
        });
    }
}" x-init="fetchData()">
    <div x-show="isLoading">
        <div class="text-gray-800">loading...</div>
    </div>
    <template x-if="rows">
        <div>show the fetched data</div>
    </template>
    <div x-show="hasError">
        <div class="text-red-500">Error occurred while fetching data</div>
    </div>
</div>
```

**Comment by ryangjchandler** [comment]
Try changing the functions to short closures instead of regular anonymous functions.

**Reply by viezel** [reply]
Wow, that actually worked. How can this have any change to the runtime?

**Reply by ryangjchandler** [reply]
@viezel it's to do with the `this` context. With a regular anon. function, Alpine's `this` binding is lost, so changing `this.isLoading` won't work. When using short closures / arrow functions, the `this` context is inherited from the calling scope, in this case the `fetchData()` function.

Alpine will bind to the `this` on `fetchData()`, the short closures will inherit that scope and therefore any changes made to `this` will be reflected / reactive.

**Reply by viezel** [reply]
I see @ryangjchandler. Thank you for the answer
